src/hooks/stores/useSyncStore.ts

```diff
@@ -0,0 +1,102 @@
+import { useMemo } from "react";
+import { create } from "zustand";
+
+type Progress = {
+  completed: number;
+  total: number;
+};
+
+type ActiveSync = {
+  progress: Progress;
+  syncIsFinished: boolean;
+};
+
+type AllSyncs = Record<string, ActiveSync>;
+
+type SyncStore = {
+  allSyncs: AllSyncs;
+  setIndivdualSync: (deviceId: string, progress?: Progress) => void;
+  resetSyncStore: () => void;
+};
+
+const emptyProgress: Progress = {
+  completed: 0,
+  total: 0,
+};
+
+const useSyncStore = create<SyncStore>()((set) => ({
```

Comment:
as an attempt to apply some helpful tips from https://tkdodo.eu/blog/working-with-zustand#separate-actions-from-state, think this can be reorganized a little so that defining additional actions is trivial and doesn't require a custom hook for individual actions (which should be stable function references to begin with). for example:

```ts
const useSyncStore = create<SyncStore>((set) => ({
  allSyncs: {},
  // Renamed the actions for the sake of the example to reduce verbosity; not necessary
  actions: {
    setSync: ...,
    reset: ...,
  }
}))

const useSyncs = (id) => {
  return useSyncStore(state => state.allSyncs)
}

// This is the main thing
// having a single hook for actions is helpful 
// and works because it should be a stable object 
// when using properly defined updater functions
const useSyncActions = ()  => {
  return useSyncStore(state => state.actions)
}
```

This way your components will always have access to the relevant actions, even if they need to use all of them:

```tsx
const App = () => {
  const syncs = useSyncs()
  const { reset, setSync } = useSyncActions()
  
  
  return ( ... )
}
```

Reply:
Yeah, this does make sense. But I specifically designed the `useIndividualSync` to just control 1 single sync object. Specifically it initializes a sync object if it does not exist.

Based on the suggested implementation, it require the developer to:

```ts
const myComponent = ({deviceId}) => {
    const syncs = useSyncs()
    const { reset, setSync } = useSyncActions()
    
    const thisSync = syncs[deviceId]
 
    if(!thisSync){
        setSync(deviceId)
    }
    
   // update Progress:
   setSync(deviceId, {total, completed})
    
}
```

As opposed to my implementation, which encapsulates alot of that boilerplate

```ts
const myComponent = ({deviceId}) => {
    const [thisSync, setThisSync] = useIndividualSync(deviceId)

    // update Progress:
   setThisSync({total, completed})
    
    
}
```

Reply:
true! although you can implement the individual hook using the other two hooks in the suggestion and still use that as you have it.

mostly just commenting about consolidating the actions, which can be be used in your custom hooks if desired :)

Reply:
Addressed: https://github.com/digidem/mapeo-mobile-shell/pull/11/commits/6f62ffe2bc4b0bfb3478f8cc5a242158098bb4de